modules/openxBidAdapter.js

```diff
@@ -230,4 +262,72 @@ function buildOXRequest(bids, oxParams, delDomain) {
   };
 }
 
+function buildOXVideoRequest(bids) {
+  if (bids.length === 0) {
+    return [];
+  }
+  return bids.map(function(bid) {
+    let url = 'http://' + bid.params.delDomain + '/v/1.0/avjp';
+    let oxVideoParams = generateVideoParameters(bid);
+    return {
+      method: 'GET',
+      url: url,
+      data: oxVideoParams,
+      payload: {'bid': bid, 'startTime': new Date()}
+    };
+  });
+}
+
+function generateVideoParameters(bid) {
+  let oxVideo = bid.params.video;
+  let oxVideoParams = {
+    auid: bid.params.unit,
+    url: oxVideo.url,
+    jsonp: 'json'
+  };
+
+  if (oxVideo.vtest) {
+    oxVideoParams.vtest = oxVideo.vtest;
+  }
+  if (oxVideo.be) {
+    oxVideoParams.be = oxVideo.be;
+  }
+
+  return oxVideoParams
+}
+
+function createVideoBidResponses(response, {bid, startTime}) {
+  let bidResponses = [];
+
+  if (response !== undefined && response.cache_key !== '') {
+    let bidResponse = {};
+    bidResponse.requestId = bid.bidId;
+    bidResponse.bidderCode = BIDDER_CODE;
+    // default 5 mins
+    bidResponse.ttl = 300;
+    // true is net, false is gross
+    bidResponse.netRevenue = true;
+    // waiting for this to be exposed
+    bidResponse.currency = 'USD';
+
+    if (response.pub_rev) {
+      bidResponse.cpm = Number(response.pub_rev);
+    } else {
+      bidResponse.cpm = 0;
+    }
+
+    bidResponse.width = bid.sizes[0];
+    bidResponse.height = bid.sizes[1];
+
+    bidResponse.openx = {
+      ff: response.cache_key,
+      oxcolo: response.per_colo_domain,
+      oxph: response.ph
+    };
+    bidResponses.push(bidResponse);
+  }
+
+  return bidResponses;
```

Comment:
i wonder if this could be much more simplified, e.g. 
```javascript
return [{
requestId: bid.bidId, 
bidderCode: BIDDER_CODE, 
ttl: 300, 
netRevenue: true, 
currency: 'USD', 
cpm : Number(response.pub_rev) || 0, 
width: bid.sizes && bid.sizes[0] || 0, 
height: bid.sizes && bid.sizes[1] || 0, 
openx: {ff: response.cache_key, oxcolo: response.per_colo_domain, oxph: response.ph}}
]
```